src/bun.js/webcore/prompt.zig

```diff
@@ -152,6 +152,130 @@ fn confirm(globalObject: *jsc.JSGlobalObject, callframe: *jsc.CallFrame) bun.JSE
 }
 
 pub const prompt = struct {
+    const KEY_CTRL_C = 3;
+    const KEY_CTRL_D = 4;
+    const KEY_BACKSPACE = 8;
+    const KEY_TAB = 9;
+    const KEY_ENTER = 13;
+    const KEY_ESC = 27;
+    const KEY_DEL = 127;
+
+    fn handleBackspace(input: *std.ArrayList(u8), cursor_index: *usize) void {
+        if (cursor_index.* > 0) {
+            const old_cursor_index = cursor_index.*;
+            const prev_codepoint_start = utf8Prev(input.items, old_cursor_index);
+
+            if (prev_codepoint_start) |start| {
+                var i: usize = 0;
+                while (i < old_cursor_index - start) : (i += 1) {
+                    _ = input.orderedRemove(start);
+                }
+                cursor_index.* = start;
+            } else {
+                _ = input.orderedRemove(old_cursor_index - 1);
+                cursor_index.* -= 1;
+            }
+        }
+    }
+
+    fn utf8Prev(slice: []const u8, index: usize) ?usize {
+        if (index == 0) return null;
+        var i = index - 1;
+        // Search backward for the start byte of a codepoint.
+        // A continuation byte starts with 0b10xxxxxx.
+        while (i > 0 and (slice[i] & 0b11000000) == 0b10000000) {
+            i -= 1;
+        }
+        // If we found a start byte, return its index.
+        // This handles ASCII (0xxxxxxx) and multibyte start bytes (11xxxxxx).
+        // If we stopped at 0, it's either a valid start byte or an invalid continuation byte.
+        // We return i, and the caller's logic will handle the deletion.
+        return i;
+    }
+
+    fn utf8Next(slice: []const u8, index: usize) ?usize {
+        if (index >= slice.len) return null;
+        // Use the project's internal function to get the byte length of the codepoint.
+        const len = bun.strings.utf8ByteSequenceLength(slice[index]);
+        const next_index = index + len;
+        if (next_index > slice.len) return null;
+        return next_index;
+    }
+
+    fn fullRedraw(stdout_writer: anytype, input: *std.ArrayList(u8), cursor_index: usize, prompt_width: usize) !void {
+        // 1. Move cursor to the start of the input area using absolute positioning (column prompt_width + 1).
+        _ = stdout_writer.print("\x1b[{d}G", .{prompt_width + 1}) catch {};
```

Comment:
why `catch {}`? Looks like it would be okay to return writer errors from this function